### PR TITLE
Cleanup fields, refactorings

### LINKED ISSUE #9: Fix object name fields in logs
Handlers are not using the right name functions.

### PR DESCRIPTION
**Pull request by ash2k**
Fixes #9. Also some refactoring and cleanup. See each commit individually.

### CODE REVIEW

File: controlled_resource_handler.go [hunk 1]
```diff
@@ -9,107 +9,105 @@ import (
...
 	}
-	return name, meta.GetNamespace()
+	return name, obj.GetNamespace()
 }
 
-func (g *ControlledResourceHandler) loggerForObj(obj interface{}) *zap.Logger {
-	logger := g.Logger
-	metaObj, ok := obj.(meta_v1.Object)
-	if ok { // This is conditional to deal with tombstones on delete event
-		logger = logger.With(logz.Namespace(metaObj), g.ZapNameField(metaObj.GetName()))
```
Comment (ash2k) [inline controlled_resource_handler.go:112]: This is the bug - it is using the name function for a controller with a name of a controlled object. Mess in the logs.

File: logz/logz.go [hunk 2]
```diff
@@ -9,16 +9,28 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
-func Gvk(gvk schema.GroupVersionKind) zapcore.Field {
```
Comment (ash2k) [inline logz/logz.go:12]: Group kinds for both controller and controlled object are mentioned in the logs using fixed field names. Same with object names.
No need to log versions, only group and kind are needed.

File: logz/logz.go [hunk 3]
```diff
@@ -9,16 +9,28 @@ import (
...
+func Controller(obj meta_v1.Object) zapcore.Field {
+	return ControllerName(obj.GetName())
+}
+
+func ControllerName(name string) zapcore.Field {
+	return zap.String("ctrl_name", name)
+}
+
+func ControllerGk(gk schema.GroupKind) zapcore.Field {
+	return zap.Stringer("ctrl_gk", &gk)
```
Comment (ash2k) [inline logz/logz.go:21]: GVK implements String on value receiver but GK on pointer receiver :tableflip:

File: lookup_handler.go [hunk 4]
```diff
@@ -8,24 +8,27 @@ import (
...
 	objs, err := e.Lookup(obj.(runtime.Object))
 	if err != nil {
-		logger.Error("Failed to get objects from index", zap.Error(err))
+		logger.Error("Failed to lookup objects", zap.Error(err))
 		return
 	}
 	for _, o := range objs {
 		metaobj := o.(meta_v1.Object)
+		logger.With(logz.Controller(metaobj)).Sugar().
+			Infof("Enqueuing controller object because controlled object was %s", addUpdateDelete)
```
Comment (ash2k) [inline lookup_handler.go:31]: This now logs for each object and does not log if there are no objects to enqueue.

**Review by amckague — APPROVED** [review]

**Review by nickhiggs — APPROVED** [review]
lgtm

**Pull request merged**